hmpps-prisoner-search-indexer/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/indexer/listeners/DomainEventPublisherListener.kt

```diff
@@ -19,24 +19,17 @@ class DomainEventPublisherListener(
   private val hmppsDomainTopic by lazy {
     hmppsQueueService.findByTopicId("hmppseventtopic") ?: throw IllegalStateException("hmppseventtopic not found")
   }
-  private val topicArn by lazy { hmppsDomainTopic.arn }
-  private val topicSnsClient by lazy { hmppsDomainTopic.snsClient }
 
   @SqsListener("publish", factory = "hmppsQueueContainerFactoryProxy")
   @WithSpan(value = "syscon-devs-hmpps_prisoner_search_publish_queue", kind = SpanKind.SERVER)
   fun publish(eventJson: String) {
     val event = fromJson<DomainEvent>(eventJson)
 
-    val request = PublishRequest.builder()
-      .topicArn(topicArn)
-      .message(event.body)
-      .messageAttributes(
-        mapOf(
-          "eventType" to MessageAttributeValue.builder().dataType("String").stringValue(event.eventType).build(),
-        ),
-      ).build()
-
-    topicSnsClient.publish(request).join()
+    hmppsDomainTopic.publish(
+      eventType = event.eventType,
+      event = event.body,
+      retryPolicy = NeverRetryPolicy(),
+    )
```

Comment:
Are you sure that this shouldn't retry? If the openSearch update succeeds then when the prisoner event message is retried it will find that nothing has changed - so another domain event won't be published.